### PR TITLE
Test for kron and bcast minus

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by andreasnoack — APPROVED** [review]

File: test/sparse/sparsevector.jl [hunk 1]
```diff
@@ -1000,6 +1000,9 @@ let testdims = ((5,10), (20,12), (25,30))
         @test Vector(kron(x,y)) == kron(Vector(x), Vector(y))
         @test Vector(kron(Vector(x),y)) == kron(Vector(x), Vector(y))
         @test Vector(kron(x,Vector(y))) == kron(Vector(x), Vector(y))
+        # test different types
+        z = convert(SparseVector{Float16, Int8}, y)
+        @test Vector(kron(x,y)) == kron(Vector(x), Vector(y))
```
Comment (jebej) [inline test/sparse/sparsevector.jl]: `z` is not being used here

File: test/sparse/higherorderfns.jl [hunk 2]
```diff
@@ -187,6 +188,10 @@ end
             broadcast!(+, Z, X, Y); Z = sparse(fZ) # warmup for @allocated
             @test (@allocated broadcast!(+, Z, X, Y)) == 0
             @test broadcast!(+, Z, X, Y) == sparse(broadcast!(+, fZ, fX, fY))
+            broadcast!(-, fZ, fX, fY); Z = sparse(fZ)
+            broadcast!(-, Z, X, Y); Z = sparse(fZ) # warmup for @allocated
+            @test (@allocated broadcast!(-, Z, X, Y)) == 0
+            @test broadcast!(-, Z, X, Y) == sparse(broadcast!(-, fZ, fX, fY))
```
Comment (Sacha0) [inline test/sparse/higherorderfns.jl]: These tests broadcasting `-` exercise the same code paths as those for `+`; together `+` and `*` cover the full set of code paths. Perhaps nix these?

**Comment by kshyatt** [comment]
That's not true, or we have duplicate code. Look at the coverage for this
file.

File: test/sparse/higherorderfns.jl (hunk 2 above)
Comment (kshyatt) [reply test/sparse/higherorderfns.jl]: [This](https://codecov.io/gh/JuliaLang/julia/src/55ec9fea4b25d785ab7bf4f0a9c23676c570eb18/base/sparse/higherorderfns.jl#L175) is what I was trying to cover here

**Comment by Sacha0** [comment]
Could you point me to the coverage data you have in mind? Thanks!

**Comment by fredrikekre** [comment]
https://codecov.io/gh/JuliaLang/julia/src/master/base/sparse/higherorderfns.jl#L175

File: test/sparse/higherorderfns.jl (hunk 2 above)
Comment (Sacha0) [reply test/sparse/higherorderfns.jl]: To spot test that line, try
```julia
a, b = sprand(5, 0.5),  sprand(5, 0.5)
@test broadcast(-, a, b) == broadcast(-, Vector(a), Vector(b))
```
The tests above are quite heavy and indeed exercise the same code paths as the `+` tests. Best!

Comment (kshyatt) [reply test/sparse/higherorderfns.jl]: Cheers! Thanks for the explanation.

Comment (kshyatt) [reply test/sparse/higherorderfns.jl]: actually, wait a second ... why isn't line 176 in this file catching this path?

Comment (Sacha0) [reply test/sparse/higherorderfns.jl]: How does one see this pull request's measured impact on coverage? The `broadcast(-, X, Y)` test you added seems like it should do the trick, as it should generate the following call
```julia
julia> @which broadcast(-, SparseVector{Float32,Int32}(sprand(1, 1.0)), sprand(1, 1.0))
broadcast(::typeof(-), x::SparseVector, y::SparseVector) in Base.SparseArrays.HigherOrderFns at sparse/higherorderfns.jl:175
```
which should cover the line you would like to cover.

In any case, I would remove the `broadcast(-, X, Y)` addition in favor of the spot test above, as the `broadcast(-, X, Y)` line is quite heavy and primarily exercises the same code as the adjacent `broadcast(+, X, Y)`. Best!

**Comment by Sacha0** [comment]
> In any case, I would remove the broadcast(-, X, Y) addition in favor of the spot test above, as the broadcast(-, X, Y) line is quite heavy and primarily exercises the same code as the adjacent broadcast(+, X, Y). Best!

The preceding review comment was not addressed? The `broadcast(-, X, Y)` line tests around sixty-four argument combinations, exercising essentially the same code as the immediately preceding `broadcast(+, X, Y)` line.

**Comment by kshyatt** [comment]
Sorry about that, @Sacha0, it's my fault. I'll do my best to fix it today, but if I don't get around to it, anybody should feel free!

**Pull request merged**